pypeit/reduce.py

```diff
@@ -645,60 +660,115 @@ def load_skyregions(self, skymask_init):
-
         Args:
-            sobjs (:class:`pypeit.specobjs.SpecObjs`):
             basename (str):
-
+            mode (str):
+                "object" - Use specObjs to determine flexure correction
+                "slit" - Use waveimg and global_sky to determine flexure correction at the centre of the slit
+            specObjs (:class:`pypeit.specobjs.SpecObjs`, None):
+                Spectrally extracted objects
         """
         if self.par['flexure']['spec_method'] != 'skip':
-            # Measure
-            flex_list = flexure.spec_flexure_obj(sobjs, self.slits.slitord_id, self.reduce_bpm,
-                                                 self.par['flexure']['spec_method'],
-                                                 self.par['flexure']['spectrum'],
-                                                 mxshft=self.par['flexure']['spec_maxshift'])
-            # QA
-            flexure.spec_flexure_qa(sobjs, self.slits.slitord_id, self.reduce_bpm, basename, self.det, flex_list,
-                                    out_dir=os.path.join(self.par['rdx']['redux_path'], 'QA'))
+            # Initialise
+            slitSpecs = None
+            gd_slits = np.logical_not(self.reduce_bpm)
+            wvimg = self.waveimg.copy() if update_waveimg else None
+            # Perform some checks
+            if mode == "object" and specObjs is None:
+                msgs.warn("No spectral extractions provided for flexure, using slit center instead")
+                mode = "slit"
+            elif mode not in ["object", "slit"]:
+                msgs.warn("mode must be 'object' or 'slit'. Assuming 'slit'.")
+                mode = "slit"
+            # Prepare a list of slit spectra, if required.
+            if mode == "slit":
+                # TODO :: Need to think about spatial flexure - is the appropriate spatial flexure already included in trace_spat via left/right slits?
+                trace_spat = 0.5 * (self.slits_left + self.slits_right)#/(self.slits.nspat-1)
+                trace_spec = np.arange(self.slits.nspec)#/(self.slits.nspec-1)
+                slitSpecs = []
+                for ss in range(self.slits.nslits):
+                    if not gd_slits[ss]:
+                        slitSpecs.append(None)
+                        continue
+                    slit_spat = self.slits.spat_id[ss]
+                    mask = (self.slitmask == slit_spat)
+                    box_denom = moment1d(self.waveimg * mask > 0.0, trace_spat[:, ss], 2, row=trace_spec)[0]
+                    wghts = (box_denom + (box_denom == 0.0))
+                    slit_sky = moment1d(self.global_sky * mask, trace_spat[:, ss], 2, row=trace_spec)[0] / wghts
+                    # Denom is computed in case the trace goes off the edge of the image
+                    slit_wave = moment1d(self.waveimg * mask, trace_spat[:, ss], 2, row=trace_spec)[0] / wghts
+                    slitSpecs.append(xspectrum1d.XSpectrum1D.from_tuple((slit_wave, slit_sky)))
```

Comment:
I really dislike adding an XSpectr1d dependency here to this routine.

Reply:
While I appreciate the concern, I don't think this needs to be addressed with this PR - it's a separate issue that needs more thought. That said, it's a fairly localised dependency (it's only really needed in `flexure.spec_flex_shift`). I've added a TODO.